VideoPlayer.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  VideoPlayer.swift
+//  OSAT-VideoCompositor
+//
+//  Created by Rohit Sharma on 13/10/22.
+//
+
+import AVFoundation
+import UIKit
+
+open class VideoPlayer: UIView {
+    private var player: AVPlayer?
+    private var playerLayer: AVPlayerLayer?
+    private var url: URL?
+    
+    private lazy var playbutton: UIButton = {
+        let btn = UIButton(frame: .zero)
+        btn.translatesAutoresizingMaskIntoConstraints = false
+        btn.addTarget(self, action: #selector(handlePlayButtonAction(_:)), for: .touchUpInside)
+        return btn
+    }()
+    
+    public convenience init() {
+        self.init(frame: .zero)
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setVideoPlayer()
+    }
```

Comment:
What if the user wants to pass in frame for the view

Reply:
We can use default arguments instead then

Reply:
<img width="1094" alt="Screenshot 2022-11-11 at 8 34 04 PM" src="https://user-images.githubusercontent.com/78854679/201367476-c6200081-4ac2-4dd0-89aa-1353fc333a2b.png">
with super.init() issue will be this error

Reply:
We can have a function like
public init(frame: CGRect = .zero) {
super.init(frame: frame)
setVideoPlayer()
}
Instead of functions in line 23 and 27